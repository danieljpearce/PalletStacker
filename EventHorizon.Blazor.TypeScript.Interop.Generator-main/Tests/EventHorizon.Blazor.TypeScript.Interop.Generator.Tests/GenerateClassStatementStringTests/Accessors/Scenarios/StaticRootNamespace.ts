declare module Static {
    export interface IInterfaceResponseType { }

    export class ExampleClass {
        static get InterfaceResponseType(): IInterfaceResponseType;
    }
}
