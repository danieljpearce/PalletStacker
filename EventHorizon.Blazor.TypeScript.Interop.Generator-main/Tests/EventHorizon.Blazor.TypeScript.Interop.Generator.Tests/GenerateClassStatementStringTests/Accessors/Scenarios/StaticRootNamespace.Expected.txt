/// Generated - Do Not Edit
namespace Static
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClass>))]
    public class ExampleClass : CachedEntityObject
    {
        #region Static Accessors
        private static IInterfaceResponseTypeCachedEntity __InterfaceResponseType;
        public static IInterfaceResponseTypeCachedEntity InterfaceResponseType
        {
            get
            {
            if(__InterfaceResponseType == null)
            {
                __InterfaceResponseType = EventHorizonBlazorInterop.GetClass<IInterfaceResponseTypeCachedEntity>(
                    "Static",
                    "ExampleClass.InterfaceResponseType",
                    (entity) =>
                    {
                        return new IInterfaceResponseTypeCachedEntity() { ___guid = entity.___guid };
                    }
                );
            }
            return __InterfaceResponseType;
            }
        }
        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors

        #endregion

        #region Properties

        #endregion
        
        #region Constructor
        public ExampleClass() : base() { }

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
            ___guid = entity.___guid;
        }


        #endregion

        #region Methods

        #endregion
    }
}